### ISSUE TITLE
about  '--train_decoder'

### ISSUE THREAD

**Issue by huangfaan** [eladrich/pixel2style2pixel#167]
self.parser.add_argument('--train_decoder', default=False, type=bool, help='Whether to train the decoder model')

train my own dataset, does  '--train_decoder' need to be set as True？？？

**Comment by yuval-alaluf** [comment]
In all our experiments, we kept the decoder fixed. In your experiments you need to make sure that you have a pretrained StyleGAN on your domain.

**Comment by huangfaan** [comment]
> In all our experiments, we kept the decoder fixed. In your experiments you need to make sure that you have a pretrained StyleGAN on your domain.

I just want to train the ffhq dataset to generate 256*256 imgs, I don't have any pretrained StyleGAN except the official stylegan2-ffhq1024-config-f.pt. So I need to reset '--train_decoder' as False ???  Besides, I run it as the '--train_decoder' is True, but the loss is inf after step is 400. 

Metrics for train, step 350
	loss_l2 =  61012426752.0
	loss_lpips =  1.1697169542312622
	loss =  61012426752.0
Metrics for train, step 400
	loss_l2 =  inf
	loss_lpips =  0.7947787642478943
	loss =  inf
Metrics for train, step 450
	loss_l2 =  inf
	loss_lpips =  0.7910941243171692
	loss =  inf
Metrics for train, step 500
	loss_l2 =  inf
	loss_lpips =  0.7548491954803467
	loss =  inf
Metrics for train, step 550
	loss_l2 =  inf
	loss_lpips =  0.7669035196304321
	loss =  inf
Metrics for train, step 600
	loss_l2 =  inf
	loss_lpips =  0.7762438058853149
	loss =  inf

**Comment by huangfaan** [comment]
> In all our experiments, we kept the decoder fixed. In your experiments you need to make sure that you have a pretrained StyleGAN on your domain.

I just want to train the ffhq dataset to generate 256*256 imgs, I don't have any pretrained StyleGAN except the official stylegan2-ffhq1024-config-f.pt. So do I need to reset '--train_decoder' as False ???

**Comment by huangfaan** [comment]
> In all our experiments, we kept the decoder fixed. In your experiments you need to make sure that you have a pretrained StyleGAN on your domain.

I just want to train the ffhq dataset to generate 256*256 imgs, I don't have any pretrained StyleGAN except the official stylegan2-ffhq1024-config-f.pt. So do I need to reset '--train_decoder' as True ???

**Comment by yuval-alaluf** [comment]
You do not need to use the flag `--train-decoder.` since the domain of your data (ffhq) is the same as the domain the StyleGAN was trained on.

**Comment by huangfaan** [comment]
> You do not need to use the flag `--train-decoder.` since the domain of your data (ffhq) is the same as the domain the StyleGAN was trained on.

ok, thanks,  but if I want to train other dataset(256*256), and I have no pretrained StyleGAN on my domain, so I need to use the flag `--train-decoder, all right??

**Comment by yuval-alaluf** [comment]
Not exactly. You need to train a StyleGAN generator on your domain. An assumption of pSp is that you have a pre-trained generator. As I said before, in all our experiments we did not train the generator. We added the option to train the generator for those interested to play around with it.

**Comment by huangfaan** [comment]
ok, I understand, actually I have a StyleGAN generator on my domain, so I'll try to train my pSp. Thank you very much.
